packages/react-native-tab-view/src/__tests__/index.test.tsx

```diff
@@ -49,3 +102,71 @@ test('switches tabs when the tab header for another route is pressed', () => {
   fireEvent.press(getByLabelText('Second'), {} as GestureResponderEvent);
   expect(getByTestId('route2')).toBeTruthy();
 });
+
+test('passes onPageSelected prop from TabView to Pager', () => {
+  const pagerMock = require('../Pager');
+  render(<ComponentWithTabView />);
+
+  // Verify onPageSelected was passed to the Pager
+  expect(pagerMock.__mockOnPageSelected).toBeDefined();
+});
```

Comment:
Avoid testing implementation details. They should only cover public API behavior.